Eatery/Views/Eateries/ExtendedFilterButton.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  ExpandedFilterButton.swift
+//  Eatery
+//
+//  Created by Sergio Diaz on 12/24/20.
+//  Copyright © 2020 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+enum ExtendedFilterButtonState {
+    case inactive, hightolow, lowtohigh
+}
+
+class ExtendedFilterButton: UIButton {
+
+    var topRect: UIView!
+    var middleRect: UIView!
+    var bottomRect: UIView!
+
+    var topSpacingRect: UIView!
+    var bottomSpacingRect: UIView!
+
+    let inactiveColor: UIColor
+    let activeColor: UIColor
+    var filterState: ExtendedFilterButtonState
```

Comment:
Private fields if possible

Reply:
Filter state needs to be accessed in another class.  Should I leave it public or add a getter method?

Reply:
Can you try `private(set)`?